lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -3901,11 +3901,11 @@ class TypeReprAvailabilityWalker : public ASTWalker {
 
     bool foundAnyIssues = false;
 
-    if (auto *GTR = dyn_cast<GenericIdentTypeRepr>(declRefTR)) {
+    if (declRefTR->hasGenericArgList()) {
       auto genericFlags = flags;
       genericFlags -= DeclAvailabilityFlag::AllowPotentiallyUnavailableProtocol;
 
-      for (auto *genericArg : GTR->getGenericArgs()) {
+      for (auto *genericArg : declRefTR->getGenericArgs()) {
```

Comment:
Did we previously miss the generic args for MemberTypeRepr? Could we add a test for that?

Reply:
Silly me! Well noticed, will do.